src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -1001,13 +996,135 @@ pub fn get_mounts(paths: &HashMap<String, String>) -> Result<HashMap<String, Str
     Ok(m)
 }
 
+#[inline]
 fn new_cgroup(h: Box<dyn cgroups::Hierarchy>, path: &str) -> Result<Cgroup> {
     let valid_path = path.trim_start_matches('/').to_string();
     cgroups::Cgroup::new(h, valid_path.as_str()).map_err(anyhow::Error::from)
 }
 
+#[inline]
+fn load_cgroup(h: Box<dyn cgroups::Hierarchy>, path: &str) -> Cgroup {
+    let valid_path = path.trim_start_matches('/').to_string();
+    cgroups::Cgroup::load(h, valid_path.as_str())
+}
+
 impl Manager {
-    pub fn new(cpath: &str) -> Result<Self> {
+    pub fn new(
+        cpath: &str,
+        spec: &Spec,
+        devcg_info: Option<Arc<RwLock<DevicesCgroupInfo>>>,
+    ) -> Result<Self> {
+        let (paths, mounts) = Self::get_paths_and_mounts(cpath).context("Get paths and mounts")?;
+
+        // Do not expect poisoning lock
+        let mut devices_group_info = devcg_info.as_ref().map(|i| i.write().unwrap());
+        let pod_cgroup: Option<Cgroup>;
+
+        if let Some(devices_group_info) = devices_group_info.as_mut() {
+            // Cgroup path of parent of container
+            let pod_cpath = PathBuf::from(cpath)
```

Comment:
what  about  container cgroup didn't have a parent path? such as "/container_cgroup"?

Reply:
Fixed. Device cgroup won't work if cgroup path is "/" or "/{anything}".